trieste/acquisition/multi_objective/partition.py

```diff
@@ -11,42 +11,67 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-""" This module contains functions and classes for Pareto based multi-objective optimization. """
+"""
+This module contains functions of different methods for
+partitioning the dominated/non-dominated region in multi-objective optimization problems
+ """
 from __future__ import annotations
 
+from abc import ABC
 from dataclasses import dataclass
 
 import tensorflow as tf
-from typing_extensions import Final
 
-from ..types import TensorType
-from .misc import DEFAULTS
+from ...types import TensorType
+from ...utils.misc import DEFAULTS
+from .dominance import non_dominated
 
 
-def non_dominated(observations: TensorType) -> tuple[TensorType, TensorType]:
+def prepare_default_non_dominated_partition_bounds(
+    observations: TensorType, anti_reference: TensorType, reference: TensorType
+) -> tuple[TensorType, TensorType]:
     """
-    Computes the non-dominated set for a set of data points.
-    if there are duplicate point(s) in the non-dominated set, this function will return
-    as it is without removing the duplicate.
+    Prepare the default non-dominated partition boundary for acquisition function usage.
+
+    :param observations: The observations for all objectives, with shape [N, D].
+    :param anti_reference: a worst point to use with shape [D].
+        Defines the lower bound of the hypercell
+    :param reference: a reference point to use, with shape [D].
+        Defines the upper bound of the hypervolume.
+        Should be equal to or bigger than the anti-ideal point of the Pareto set.
+        For comparing results across runs, the same reference point must be used.
+    :return: lower, upper bounds of the partitioned cell
+    :raise ValueError (or `tf.errors.InvalidArgumentError`): If ``reference`` has an invalid
+        shape.
+    """
+    if observations.shape[-1] > 2:
+        return DividedAndConquerNonDominated(observations).partition_bounds(
+            anti_reference, reference
+        )
+    elif observations.shape[-1] == 2:
+        return ExactPartition2dNonDominated(observations).partition_bounds(
+            anti_reference, reference
+        )
+    else:
+        raise ValueError(f"observations: {observations} not understood")
 
-    :param observations: set of points with shape [N,D]
-    :return: tf.Tensor of the non-dominated set [P,D] and the degree of dominance [N],
-        P is the number of points in pareto front
-        dominances gives the number of dominating points for each data point
 
+class Partition(ABC):
```

Comment:
More from an understanding perspective.

Maybe we can remove the (general) partition class and keep the `NonDominatedPartition` and `DominatedPartition` . This may make sense from an understanding perspective as `NonDominatedPartition` is always partition the non-dominated regions while `DominatedPartition` is doing the opposite, which gives clear indication of its functionality. pls leme know if its still better to be even more concise (i.e., avoid any empty ABC) 

`partition_bounds` is generic but unfortunately different class may not share same input args, so decided not put in ABC.